docs/csharp/fundamentals/functional/deconstruct.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Deconstructing tuples and other types
+description: Learn how to deconstruct tuples and other types.
+ms.technology: csharp-fundamentals
+ms.date: 03/22/2021
+---
+# Deconstructing tuples and other types
+
+A tuple provides a lightweight way to retrieve multiple values from a method call. But once you retrieve the tuple, you have to handle its individual elements. Working on an element-by-element basis is cumbersome, as the following example shows. The `QueryCityData` method returns a three-tuple, and each of its elements is assigned to a variable in a separate operation.
+
+:::code language="csharp" source="./snippets/deconstructing-tuples/deconstruct-tuple1.cs":::
+
+Retrieving multiple field and property values from an object can be equally cumbersome: you must assign a field or property value to a variable on a member-by-member basis.
+
+In C# 7.0 and later, you can retrieve multiple elements from a tuple or retrieve multiple field, property, and computed values from an object in a single *deconstruct* operation. To deconstruct a tuple, you assign its elements to individual variables. When you deconstruct an object, you assign selected values to individual variables.
+
+## Deconstructing a tuple
```

Comment:
Is it useful to start each header with the same word? Some of the other headers will wrap.